allauth/socialaccount/providers/base/provider.py

```diff
@@ -261,4 +273,10 @@ def __str__(self):
         So we have this method `to_str` that can be overridden in a conventional
         fashion, without having to worry about it.
         """
-        return self.get_brand()["name"]
+        ed = self.account.extra_data
+        return (
```

Comment:
Given that this method is the default implementation, and that `extra_data` is completely unstructured, I think we need to be a bit cautious here as to what data we can expect:

- Add a `if isinstance(ed, dict)` before invoking `ed.get()`.
- When doing `ed.get("email")`, check if the result is really a `str`.

Reply:
Very good point.